view/frontend/web/js/view/payment/method-renderer/omise-offsite-installment-method.js

```diff
@@ -69,6 +69,29 @@ define(
                 return priceUtils.formatPrice(amount, quote.getPriceFormat());
             },
 
+            /**
+             * Format Price
+             * 
+             * @param {string} option Name of option to be verified
+             * @return {boolean}
+             */
+            isInstallmentOptionActive: function (option) {
+                return parseInt(window.checkoutConfig.installment_config[option]);
+            },
+
+            /**
+             * Format Price
+             * 
+             * @param {string} option Name of option to be verified
+             * @return {boolean}
+             */
+            isAnyInstallmentOptionActive: function () {
+                return parseInt(window.checkoutConfig.installment_config['omise_offsite_installment_kbank'])
```

Comment:
A lot of this feels very rigid and inflexible. Would it not be better to change to a more flexible approach approach given that we will have the instalment options coming from Capabilities? At least moving to an architecture that is some way towards accommodating this?

Reply:
we need to move from capabilities, all coming changes will be hardcoded.

Reply:
We absolutely do not want to move from capabilities. It tells us what the merchant's account can do. We don't want to be offering them payment instalment options not supported by their account. The best way to go for now is to pull the available instalment options from Capabilities (they are already there), and create some means to switch those available ones on/off from admin in Magento - as has been requested. We should implement this in such a way that the mechanism for switching off individual instalment methods can either be switched off easily (as the switching on/off may eventually move to the Omise dashboard side), or architected in such a way that it would be easy to change the manner in which they are switch on/off (i.e. entirely decouple the means of configuring this from the way the switching on/off is implemented)

Reply:
well.. it will be hard coded, people does not have any other companies, like new company that is never listed here.

also, new pr that is coming will analyze capabilities and what installment options are available,
this pr only disable company.

Reply:
that is correct,

installment terms that are theoretically flexible for each customer, will be taken from capabilities,

that is different pr coming.

this pr only disable/enable company.

Reply:
but i get your point.. I went way tooooo far with saying EVERYTHING WILL BE HARDCODED